app/views/registrations/_item_row.html.erb

```diff
@@ -1,26 +1,42 @@
 <div class="row mb-3">
-  <%= item_form.label :source_id, 'Source ID', class: 'col-sm-2 col-form-label' %>
+  <%= item_form.label :barcode, class: 'col-sm-2 col-form-label' %>
   <div class="col-sm-10">
-    <%= item_form.text_field :source_id, class: 'form-control', required: true,
-                             pattern: '^.+:.+$',
-                             data: {
-                               action: "change->registration#validateSourceId invalid->registration#displayValidation",
-                               registration_target: 'sourceId',
-                               registration_tabs_target: 'requiredFormField'
-                             }
-     %>
+    <%= item_form.text_field :barcode, class: 'form-control',
+    pattern: '^(36105[0-9]{9}|2050[0-9]{7}|245[0-9]{8}|[0-9]+-[0-9]+)',
+    data: {
+      action: "change->registration-item-row#validateBarcode invalid->registration#displayValidation",
+      registration_item_row_target: 'barcode',
+    } %>
   </div>
 </div>
+
 <div class="row mb-3">
   <%= item_form.label :catkey, class: 'col-sm-2 col-form-label' %>
   <div class="col-sm-10">
     <%= item_form.text_field :catkey, class: 'form-control',
       pattern: '^[0-9]+(:[0-9]+)*$',
       data: {
-        action: "change->registration#validateCatkey invalid->registration#displayValidation",
+        action: "change->registration-item-row#validateCatkey invalid->registration#displayValidation",
+        registration_item_row_target: 'catkey',
     } %>
   </div>
 </div>
+
+<div class="row mb-3">
+  <%= item_form.label :source_id, 'Source ID', class: 'col-sm-2 col-form-label' %>
+  <div class="col-sm-10">
+    <%= item_form.text_field :source_id, class: 'form-control', required: true,
+                             pattern: '^.+:.+$',
+                             data: {
+                               action: "change->registration-item-row#validateSourceId invalid->registration#displayValidation",
+                               registration_target: 'sourceId',
```

Comment:
This is no longer a target of registration?

Reply:
no. It's now all done in `registration-item-row`